src/tiledb/cloud/bioimg/ingestion.py

```diff
@@ -0,0 +1,119 @@
+import math
+import os
+from functools import partial
+from typing import Any, Mapping, Optional, Sequence, Tuple, Union
+
+import tiledb
+from tiledb.cloud.utils import logger
+
+
+def ingest(source: Union[Sequence[str], str],
+           output: str, 
+           access_key_id: str, 
+           secret_access_key: str, 
+           *,
+           taskgraph_name: Optional[str] = None,
+           num_batches: Optional[int] = 1,
+           threads: Optional[int] = 8,
+           resources: Optional[Mapping[str, Any]] = None,
+           **kwargs):
+    
+    """The function ingests microscopy images into TileDB arrays
+
+    :param source: uri / iterable of uris of input files e.g 
+    :param output: output dir for the ingested tiledb arrays
+    :param access_key_id: AWS_ACCESS_KEY_ID
+    :param secret_access_key: AWS_SECRET_ACCESS_KEY
+    :param taskgraph_name: Optional name for taskgraph, defaults to None
+    :param num_batches: Number of graph nodes to spawn performs it sequentially if default, defaults to 1
```

Comment:
I think we want this to be higher than one? With one doesn't this mean a single node will sequentially process all images? We should use some internal heuristics, likely building one task per image and then limit the concurrency?

Reply:
Sure I had it in mind too, but I followed the simple and straightforward path. The change is easy. However I do not know if we also would like to spawn a taskgraph of 1000 nodes for a dataset of 1000 images.

Reply:
There is support for limiting the concurrency of a task graph, so you could launch 1000 independent nodes but say "only run 20 at a time". @NikolaosPapailiou can provide the configuration for that.

Reply:
Yes you can set `max_workers` in the Dag object to limit parallelism.